spec/unit/locale_spec.rb

```diff
@@ -31,6 +36,42 @@
     it "process_locale_nil" do
       expect(SgtnClient::LocaleUtil.process_locale(nil)).to eq SgtnClient::Config.configurations.default
     end
+
+
+    it "default" do
+      env = SgtnClient::Config.default_environment
+      config_lang = SgtnClient::Config.configurations[env]["default_language"]
+
+      SgtnClient::Config.configurations[env]["default_language"] = nil
+      expect(SgtnClient::LocaleUtil.get_source_locale()).to eq 'en'
+
+      SgtnClient::Config.configurations[env]["default_language"] = 'en'
+      expect(SgtnClient::LocaleUtil.get_source_locale()).to eq 'en'
+      
+      SgtnClient::Config.configurations[env]["default_language"] = 'zh-Hans'
+      expect(SgtnClient::LocaleUtil.get_source_locale()).to eq 'zh-Hans'
+
+      SgtnClient::Config.configurations[env]["default_language"] = config_lang
+    end
+
+    it "fallback_locale" do
+      env = SgtnClient::Config.default_environment
```

Comment:
I don't know how this is related to [#1460](https://github.com/vmware/singleton/issues/1460)

Reply:
get_best_locale is separated from get_cachekey and get_cachekey's return value is changed already which will have effect on get_cs, so it's associated.